### PR TITLE
[Performance] Clean and refine softmax and gemm benchmarks

### PR DESCRIPTION
**Pull request by ESI-SYD**
1. Add bgemm xetla kernel to benchmarks
2. Modify shapes cfgs, and refine softmax's integration.
3. Add code snippet(default commented out) supporting print perf in xetla's `cpp/h` files.

We still need to further investigation difference between this intergration and run xetla kernels directly.

### CODE REVIEW

File: benchmarks/xetla_benchmark/experimental-block-pointer.py [hunk 1]
```diff
@@ -0,0 +1,277 @@
+"""
+Block Pointer (Experimental)
```
Comment (chengjunlu) [inline benchmarks/xetla_benchmark/experimental-block-pointer.py:2]: Remove these comments. Clean up the code just for benchmark usage.

File: benchmarks/xetla_benchmark/experimental-block-pointer.py [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+    a = torch.randn((M, K), device='xpu', dtype=torch.float16)
+    b = torch.randn((K, N), device='xpu', dtype=torch.float16)
+    quantiles = [0.5, 0.2, 0.8]
+
+    # calculate tflops for oneDNN kernel
+    def calculate_tflops(ms):
+        return 2 * M * N * K * 1e-12 / (ms * 1e-3)
+
+    if provider == 'onednn':
+        ms, min_ms, max_ms = triton.testing.do_bench(lambda: torch.matmul(a, b), rep=100, quantiles=quantiles,
```
Comment (chengjunlu) [inline benchmarks/xetla_benchmark/experimental-block-pointer.py:259]: Change the warm up number and rep number to 10

File: benchmarks/xetla_benchmark/fused_softmax.py [hunk 3]
```diff
@@ -38,6 +38,31 @@ def naive_softmax(x):
...
+        triton.Config({'BLOCK_SIZE': 512}, num_warps=8),
+        triton.Config({'BLOCK_SIZE': 1024}, num_warps=8),
+        triton.Config({'BLOCK_SIZE': 2048}, num_warps=8),
+        triton.Config({'BLOCK_SIZE': 128}, num_warps=4),
+        triton.Config({'BLOCK_SIZE': 256}, num_warps=4),
+        triton.Config({'BLOCK_SIZE': 512}, num_warps=4),
+        triton.Config({'BLOCK_SIZE': 1024}, num_warps=4),
+        triton.Config({'BLOCK_SIZE': 2048}, num_warps=4),
+    ],
+    key=[],
```
Comment (chengjunlu) [inline benchmarks/xetla_benchmark/fused_softmax.py:64]: Please add the "BLOCK_SIZE" into the key. We should use different configuration based on different problem size used in the softmax.

File: benchmarks/xetla_kernel/bgemm/cm_kernel.cpp [hunk 4]
```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2020, Intel Corporation
```
Comment (chengjunlu) [inline benchmarks/xetla_kernel/bgemm/cm_kernel.cpp:2]: Remove this file. We don't use the cm_kernel.

File: benchmarks/xetla_benchmark/fused_softmax.py [hunk 5]
```diff
@@ -70,23 +95,19 @@ def softmax(x):
     # increasing the number of warps (`num_warps`) over which each row is distributed.
     # You will see in the next tutorial how to auto-tune this value in a more natural
     # way so you don't have to come up with manual heuristics yourself.
-    num_warps = 4
-    if BLOCK_SIZE >= 2048:
-        num_warps = 8
-    if BLOCK_SIZE >= 4096:
-        num_warps = 16
+    num_warps = 32
+    # if BLOCK_SIZE >= 2048:
```
Comment (chengjunlu) [inline benchmarks/xetla_benchmark/fused_softmax.py:99]: Remove the commented out code.

File: benchmarks/xetla_benchmark/experimental-block-pointer.py [hunk 6]
```diff
@@ -0,0 +1,277 @@
+"""
```
Comment (chengjunlu) [inline benchmarks/xetla_benchmark/experimental-block-pointer.py:1]: Change the file name. gemm_benchmark.py

File: benchmarks/xetla_benchmark/experimental-block-pointer.py [hunk 7]
```diff
@@ -0,0 +1,277 @@
...
+import triton
+import triton.language as tl
+
+import xetla_benchmark
+import xetla_benchmark.xetla_kernel as xetla_kernel
+
+
+@triton.autotune(
+    configs=[
+        triton.Config({'BLOCK_SIZE_M': 128, 'BLOCK_SIZE_N': 256, 'BLOCK_SIZE_K': 64, 'GROUP_SIZE_M': 4}, num_stages=4,
```
[the remark this answers is not in the document]
Comment (ESI-SYD) [reply benchmarks/xetla_benchmark/experimental-block-pointer.py]: We keep to use `num_stages=4` in previous gemm perf collection in private branch. And in [link](https://github.com/intel/intel-xpu-backend-for-triton/blob/llvm-target/python/tutorials/09-experimental-block-pointer.py#L102)

File: benchmarks/xetla_benchmark/fused_softmax.py [hunk 8]
```diff
@@ -38,6 +38,31 @@ def naive_softmax(x):
...
+        triton.Config({'BLOCK_SIZE': 512}, num_warps=8),
+        triton.Config({'BLOCK_SIZE': 1024}, num_warps=8),
+        triton.Config({'BLOCK_SIZE': 2048}, num_warps=8),
+        triton.Config({'BLOCK_SIZE': 128}, num_warps=4),
+        triton.Config({'BLOCK_SIZE': 256}, num_warps=4),
+        triton.Config({'BLOCK_SIZE': 512}, num_warps=4),
+        triton.Config({'BLOCK_SIZE': 1024}, num_warps=4),
+        triton.Config({'BLOCK_SIZE': 2048}, num_warps=4),
+    ],
+    key=['BLOCK_SIZE'],
```
Comment (chengjunlu) [inline benchmarks/xetla_benchmark/fused_softmax.py:64]: Opps. My bad. It should be `n_cols`. The `BLOCK_SIZE` is an autotune configuration based on the `n_cols`

**Review by chengjunlu — APPROVED** [review]

**Pull request merged**